trie-db/src/trie_codec.rs

```diff
@@ -128,17 +118,29 @@ impl<C: NodeCodec> EncoderStackEntry<C> {
 					let empty_child = ChildReference::Inline(C::HashOut::default(), 0);
 					C::extension_node(partial.right_iter(), partial.len(), empty_child)
 				},
-			NodePlan::Branch { value, children } => C::branch_node(
-				Self::branch_children(node_data, &children, &self.omit_children)?.iter(),
-				value.as_ref().map(|v| v.build(node_data)),
-			),
+			NodePlan::Branch { value, children } => {
+				let value = if self.omit_value {
+					value.as_ref().map(|_v| crate::node::Value::Inline(&[]))
```

Comment:
This could be simplified to:

```rust
value.is_some().then_some(NO_VALUE)
```

...not sure whether this is a little too much code golf-y or not though, so up to you. :P